### DISCUSSION TITLE
Is there a way to add a trailing slash to the URL of blog pages?

### DISCUSSION THREAD

**Discussion by flschr** [datenstrom/community#638] [Ask a question] [answered]
Is there a way to add a trailing slash to the URL of blog pages? Currently all static pages automatically do have a trailing slash and I want this for blog pages too. I couldn't find a easy fix in the blog.php, but as I'm not a developer I may be wrong.

**Comment by flschr** [comment]
Btw. I already tried this .htaccess rule, but that is causing 404 errors:

```
RewriteCond %{REQUEST_URI} !(/$|\.)
RewriteRule (.*) %{REQUEST_URI}/ [R=301,L]
```

**Comment by schulle4u** [comment]
I think this is not possible. Usually blog pages are files in a blog content folder, therefore they cannot have a trailing slash.

**Reply by flschr** [reply]
Thats a pitty but I understand that there is no way to solve this. But anyway, thanks a lot!

**Reply by markseuffert** [reply]
Have a look at datenstrom/community#239

**Reply by flschr** [reply]
Very interessting discussion. I'm not sure if you mean that I can solve my "problem" with a modification of BlogLocation / BlogNewLocation?!

**Reply by markseuffert** [reply]
The blog extension assumes that blog pages are in the same folder as the main blog page. You can change this behaviour by setting the `BlogStartLocation` in your `system/extensions/yellow-system.ini`. Now the blog extension will look anywhere in your `content` folder for blog pages.  This is done in the source code with function `getBlogPages()`. As you probably know from your website, URLs are based on the folder structure.

**Comment by markseuffert** [comment, marked as answer]
An URL with trailing slash is just a folder with a file called `page.md`. Here are some examples:

The file `content/2020-04-07-blog-example/page.md` becomes `http://website/blog-example/`.  
The file `content/2020-12-06-made-for-people/page.md` becomes `http://website/made-for-people/`.  
The file `content/post/114883/kaiserschmarrn/page.md` becomes `http://website/post/114883/kaiserschmarrn/`.

Configure the following [blog settings](https://github.com/annaesvensson/yellow-blog#settings) in file `system/extensions/yellow-system.ini`:

```
BlogStartLocation: /
BlogNewLocation: /@title/
```

Adjust the `content` folder as following:

```
├── content
│   ├── 1-blog
│   ├── 9-about
│   ├── 2020-04-07-blog-example
│   │   └── page.md
│   ├── 2020-12-06-made-for-people
│   │   └── page.md
│   ├── post
│   │   └── 114883
│   │       └── kaiserschmarrn
│   │           └── page.md
│   └── shared
├── media
└── system
```

There is one downside, each blog page needs its own folder. Tested with Datenstrom Yellow 0.8.18.

**Reply by flschr** [reply]
Thanks a lot for your explanation. I definitely will have a look on this :)

**Reply by flschr** [reply]
@markseu today I was running a short test, so I changed the settings in the `yellow-system.ini` and moved 1 article into a subfolder (see screenshot). Unfortunatly Yellow is now responding with the error message **Folder 'content/1-blog' may not contain subfolders!** when I reload my blog. Did I miss something?

**Update**: I already tried a clean installation (with the latest release). I also tried to delete all *.md files inside `content/1-blog` just to be sure that they not cause the error. But the problem still occurs.

![Screenshot_1](https://user-images.githubusercontent.com/23475184/115575125-8d206e00-a2c2-11eb-920c-84bf4e349d8f.jpg)

**Reply by markseuffert** [reply]
Just add folders to `content` not to your `content/1-blog`. Check out the examples above. 

Here's a longer explanation: You should be able to create any URL you want. Because of how the filesystem is organised, the home folder must not contain subfolders. Any other folder may contain both files and folders. Should you make a folder in your home folder, then the home page will return error 500.

See also datenstrom/community#159 and [Murphy's law](https://en.wikipedia.org/wiki/Murphy%27s_law).

**Reply by markseuffert** [reply]
Does this answer your question?

**Reply by flschr** [reply]
Yeap it does. Thanks a lot :)
